detectable/src/main/java/com/synopsys/integration/detectable/detectables/pip/parser/RequirementsFileTransformer.java

```diff
@@ -1,34 +1,133 @@
+        List<String> tokensAfterOperator = null;
+
+        // Find the operator with its index that separates a dependency from its version.
+        List<Object> operatorWithIndex = findOperatorWithIndex(formattedLine);
+        String operatorFound = (String) operatorWithIndex.get(0);
+        if (!operatorFound.isEmpty()) {
+            int operatorStartIndex = (int) operatorWithIndex.get(1);
+            int operatorEndIndex = operatorStartIndex + operatorFound.length() - 1;
 
-    private List<RequirementsFileDependency> convertEntriesToDependencyInfo(Map<String, RequirementsFileDependencyEntry> dependencyEntries) {
-        return dependencyEntries.entrySet().stream()
-            .map(entry -> new RequirementsFileDependency(entry.getKey(), requirementsFileDependencyVersionParser.parseRawVersion(entry.getValue().version)))
-            .collect(Collectors.toList());
+            // Get strings before and after operator
+            String stringBeforeOperator = formattedLine.substring(0, operatorStartIndex).trim();
+            String stringAfterOperator = formattedLine.substring(operatorEndIndex + 1).trim();
+
+            // Tokenize based on whitespace as the parser should allow special characters in version and dependency strings
+            tokensBeforeOperator = Arrays.asList(stringBeforeOperator.split(" "));
+            tokensAfterOperator = Arrays.asList(stringAfterOperator.split(" "));
+        } else {
+            // No operator found. Implies version is missing. Hence, only set tokensBeforeOperator as a tokenized version of original input line.
+            tokensBeforeOperator = Arrays.asList(formattedLine.split(" "));
+        }
+        return Arrays.asList(tokensBeforeOperator, tokensAfterOperator);
+    }
+
+    public List<Object> findOperatorWithIndex(String line) {
+        int operatorIndex;
+        List<Object> operatorWithIndex = new ArrayList<>();
+        for (String operator : OPERATORS_IN_PRIORITY_ORDER) {
+            operatorIndex = line.indexOf(operator);
+            if (operatorIndex != -1) {
+                operatorWithIndex.add(operator);
+                operatorWithIndex.add(operatorIndex);
+                return operatorWithIndex;
+            }
+        }
+        return Arrays.asList("", -1);
+    }
+
+    public String formatLine(String line) {
+        int ignoreAfterIndex;
+        String formattedLine = line.trim();
+
+        // Ignore any lines that start with hyphen as these are flags, not dependency entries (can be - or --)
+        // Flags are handled in the extractor, not during dependency parsing, as they involve pre-parsing decisions (example: -r flag)
+        if (formattedLine.startsWith("-")) {
+            return "";
+        }
+
+        // Ignore any characters that appear after the chars in the IGNORE_AFTER_CHARACTERS list
+        for (String ignoreAfterChar : IGNORE_AFTER_CHARACTERS) {
+            ignoreAfterIndex = formattedLine.indexOf(ignoreAfterChar);
```

Comment:
PIP docs mention:

> Whitespace followed by a # causes the # and the remainder of the line to be treated as a comment.

I am not sure if there are other valid reasons to have `#` as part of a line; but, seems like for it to be considered a comment not at the start of the line there must also be a leading whitespace.

Reply:
Yep I had seen that. `#` is a noise character for dependency/version extraction and doesn't have a significance in either strings as a special character. So we are safe with ignoring all characters after it as we would have to remove it either way when formatting the token.